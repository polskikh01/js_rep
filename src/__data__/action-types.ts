export const PRODUCT = {
    REQUEST: 'PRODUCT_REQUEST',
    SUCCESS: 'PRODUCT_SUCCESS',
    FAILURE: 'PRODUCT_FAILURE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};

export const ACCESSORIES = {
    REQUEST: 'ACCESSORIES_REQUEST',
    SUCCESS: 'ACCESSORIES_SUCCESS',
    FAILURE: 'ACCESSORIES_FAILURE',
};

export const COMPLETE = {
    REQUEST: 'COMPLETE_REQUEST',
    SUCCESS: 'COMPLETE_SUCCESS',
    FAILURE: 'COMPLETE_FAILURE',
};