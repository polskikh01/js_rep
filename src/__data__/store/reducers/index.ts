import { combineReducers } from 'redux';

import productReducer, {ProductState} from './product'

export type AppStore = {
    product: ProductState;
};

export default combineReducers<AppStore>({
    product: productReducer
})