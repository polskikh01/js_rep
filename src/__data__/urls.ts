import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('js_rep');

export const baseUrl = navigations['js_rep'];

export const URLs = {
    main: {
        url: navigations['link.js_rep.main'],
    },
    product: {
        url: navigations['link.js_rep.product'],
    },
    accessories: {
        url: navigations['link.js_rep.accessories'],
    },
    complete: {
        url: navigations['link.js_rep.complete'],
    },
    contacts: {
        url: navigations['link.js_rep.contacts'],
    },
    shippingAndPayment: {
        url: navigations['link.js_rep.shippingAndPayment']
    }
}