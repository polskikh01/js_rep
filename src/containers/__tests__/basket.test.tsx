import React from 'react';
import {mount} from "enzyme";
import {describe, it, expect} from '@jest/globals'
import {Provider} from 'react-redux'

import {store} from '../../__data__/store'

import Step1 from '../basket/basket';
import Step2 from '../basket/basket';

describe('Тестирование всего приложения', () => {
    it('Тестируем рендер basketStep1', () => {
        expect.assertions(1)
        const component = mount(
            <Provider store={store}>
                <Step1/>
            </Provider>
        )
        expect(component).toMatchSnapshot()
    });
    it('Тестируем рендер basketStep2', () => {
        expect.assertions(1)
        const component = mount(
            <Provider store={store}>
                <Step2/>
            </Provider>
        )
        expect(component).toMatchSnapshot()
    });
})