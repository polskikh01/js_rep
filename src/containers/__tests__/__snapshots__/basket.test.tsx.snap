// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование всего приложения Тестируем рендер basketStep1 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Basket>
    <BasketPage>
      <div
        className="page"
      >
        <Head>
          <header
            className="mHeader"
          >
            <div
              className="menuHeader"
            >
              <div
                className="menu"
              >
                <Link
                  className="logo"
                  to="/main"
                >
                  <a>
                    mock link
                  </a>
                </Link>
                <nav
                  className="navP"
                >
                  <Nav
                    link="/product"
                    title="Продукты"
                  >
                    <Link
                      className="navButt"
                      to="/product"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                  </Nav>
                  <Nav
                    link="/shippingAndPayment"
                    title="Доставка и оплата"
                  >
                    <Link
                      className="navButt"
                      to="/shippingAndPayment"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                  </Nav>
                  <Nav
                    link="/contacts"
                    title="Контакты"
                  >
                    <Link
                      className="navButt"
                      to="/contacts"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                  </Nav>
                </nav>
                <div
                  className="basket"
                >
                  <Link
                    to="/basket"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </header>
        </Head>
        <div
          className="wrap"
        >
          <Connect(Basket)
            nextStep={[Function]}
          >
            <Basket
              basket={Array []}
              getBasket={[Function]}
              loading={true}
              nextStep={[Function]}
            >
              <p>
                loading...
              </p>
            </Basket>
          </Connect(Basket)>
        </div>
        <Footer>
          <footer
            className="footerM"
          >
            <div
              className="footerContainer"
            >
              <span>
                Calabash
              </span>
              <h6>
                ©2020-2021
              </h6>
            </div>
            <div
              className="footerContainer"
            >
              <span
                className="title"
              >
                Продукты
              </span>
              <Link
                to="/product"
              >
                <a>
                  mock link
                </a>
              </Link>
              <Link
                to="/accessories"
              >
                <a>
                  mock link
                </a>
              </Link>
              <Link
                to="/complete"
              >
                <a>
                  mock link
                </a>
              </Link>
            </div>
            <div
              className="footerContainer"
            >
              <span
                className="title"
              >
                Аккаунт
              </span>
              <span
                className="fCT"
              >
                Личный кабинет
              </span>
            </div>
            <div
              className="footerContainer"
            >
              <span
                className="title"
              >
                Помощь
              </span>
              <Link
                to="/shippingAndPayment"
              >
                <a>
                  mock link
                </a>
              </Link>
              <Link
                to="/contacts"
              >
                <a>
                  mock link
                </a>
              </Link>
            </div>
          </footer>
        </Footer>
      </div>
    </BasketPage>
  </Basket>
</Provider>
`;

exports[`Тестирование всего приложения Тестируем рендер basketStep2 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Basket>
    <BasketPage>
      <div
        className="page"
      >
        <Head>
          <header
            className="mHeader"
          >
            <div
              className="menuHeader"
            >
              <div
                className="menu"
              >
                <Link
                  className="logo"
                  to="/main"
                >
                  <a>
                    mock link
                  </a>
                </Link>
                <nav
                  className="navP"
                >
                  <Nav
                    link="/product"
                    title="Продукты"
                  >
                    <Link
                      className="navButt"
                      to="/product"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                  </Nav>
                  <Nav
                    link="/shippingAndPayment"
                    title="Доставка и оплата"
                  >
                    <Link
                      className="navButt"
                      to="/shippingAndPayment"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                  </Nav>
                  <Nav
                    link="/contacts"
                    title="Контакты"
                  >
                    <Link
                      className="navButt"
                      to="/contacts"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                  </Nav>
                </nav>
                <div
                  className="basket"
                >
                  <Link
                    to="/basket"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </header>
        </Head>
        <div
          className="wrap"
        >
          <Connect(Basket)
            nextStep={[Function]}
          >
            <Basket
              basket={Array []}
              getBasket={[Function]}
              loading={true}
              nextStep={[Function]}
            >
              <p>
                loading...
              </p>
            </Basket>
          </Connect(Basket)>
        </div>
        <Footer>
          <footer
            className="footerM"
          >
            <div
              className="footerContainer"
            >
              <span>
                Calabash
              </span>
              <h6>
                ©2020-2021
              </h6>
            </div>
            <div
              className="footerContainer"
            >
              <span
                className="title"
              >
                Продукты
              </span>
              <Link
                to="/product"
              >
                <a>
                  mock link
                </a>
              </Link>
              <Link
                to="/accessories"
              >
                <a>
                  mock link
                </a>
              </Link>
              <Link
                to="/complete"
              >
                <a>
                  mock link
                </a>
              </Link>
            </div>
            <div
              className="footerContainer"
            >
              <span
                className="title"
              >
                Аккаунт
              </span>
              <span
                className="fCT"
              >
                Личный кабинет
              </span>
            </div>
            <div
              className="footerContainer"
            >
              <span
                className="title"
              >
                Помощь
              </span>
              <Link
                to="/shippingAndPayment"
              >
                <a>
                  mock link
                </a>
              </Link>
              <Link
                to="/contacts"
              >
                <a>
                  mock link
                </a>
              </Link>
            </div>
          </footer>
        </Footer>
      </div>
    </BasketPage>
  </Basket>
</Provider>
`;
